src/pages/admin/_middleware.ts

```diff
@@ -1,13 +1,13 @@
 // https://next-auth.js.org/configuration/nextjs#middleware
 import { isAdmin } from 'common/util/roles'
-import { getToken } from 'next-auth/jwt'
 import withAuth from 'next-auth/middleware'
+import { getSession } from 'next-auth/react'
 
 export default withAuth({
   callbacks: {
-    authorized: async ({ req }) => {
-      const jwt = await getToken({ req })
-      return isAdmin(jwt)
+    authorized: async () => {
+      const session = await getSession()
```

Comment:
Unfortunately we can't get the session in the middleware

I think this is the main cause for the failed next build

Reply:
I think we can still use the `await getToken` instead of using the `{ token }` from the `authorize` fn

This did work in the previous revision

Reply:
If it works - it doesn't matter :)

It was a suggestion for the failed build, which seems to have failed for a different reason.

Resolving the convo